### PR TITLE
Enable doctesting

### LINKED ISSUE #566: Add doctests
I have observed people are increasingly running into examples in the docs failing due to updates and it would be helpful to have doctests for us to be able to catch them beforehand

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes #566 and fixes https://github.com/SciML/DiffEqFlux.jl/issues/726 Fixes https://github.com/SciML/DiffEqFlux.jl/issues/731

### CODE REVIEW

**Comment by prbzrg** [comment]
Related to #566

**Comment by ChrisRackauckas** [comment]
@DhairyaLGandhi can I get help fixing the dataloaders?

**Comment by DhairyaLGandhi** [comment]
Sure. I noticed a few different issues. First is that the Dataloader takes in a tuple of data for its first argument so you want to pass on `DataLoader((X,Y), ...)`.

Another is that the concentric sphere has incorrect shapes. I also saw a mutation error coming from zygote.

File: docs/src/examples/normalizing_flows.md [hunk 1]
```diff
@@ -7,12 +7,13 @@ Now, we study a single layer neural network that can estimate the density `p_x`
...
+      OptimizationOptimJL, Distributions
 
-nn = Lux.Chain(
-    Lux.Dense(1, 3, tanh),
-    Lux.Dense(3, 1, tanh),
+nn = Flux.Chain(
+    Flux.Dense(1, 3, tanh),
+    Flux.Dense(3, 1, tanh),
 ) |> f32
 tspan = (0.0f0, 10.0f0)
```
Comment (prbzrg) [inline docs/src/examples/normalizing_flows.md:18]: Shouldn't `tspan` be `(0, 1)`?
```suggestion
tspan = (0.0f0, 1.0f0)
```

**Comment by ChrisRackauckas** [comment]
Interesting that it seems to be CNF that's causing issues on the docbuilder. That example runs fine locally (with these updates).

**Comment by ChrisRackauckas** [comment]
Isolated and fixed: https://github.com/EnzymeAD/Enzyme.jl/issues/368

**Comment by ChrisRackauckas** [comment]
yippieeeee!
